src/Toolchain.ml

```diff
@@ -335,15 +316,15 @@ let getCommand (t : PackageManager.t) bin args : Path.t * string array =
     | Opam (opam, switch) -> Opam.exec opam ~switch ~args:binArgs
     | Esy (esy, manifest) -> Esy.exec esy ~manifest ~args:binArgs
     | Global -> (Path.ofString bin, Array.of_list args)
-    | Custom { ocamllsp; dune } -> (
-      let command =
-        match bin with
-        | "ocamllsp" -> String.split_on_char ' ' ocamllsp
-        | "dune" -> String.split_on_char ' ' dune
-        | _ -> [ bin ]
-      in
-      match command with
-      | bin :: extraArgs -> (Path.ofString bin, Array.of_list (extraArgs @ args))
+    | Custom run -> (
+      match
+        run
+        |> Js.String.replace "$prog" bin
+        |> Js.String.replace "$args" (String.concat "" args)
+        |> String.split_on_char ' '
```

Comment:
I fear this will not work well with quoting :/

What about interpreting this as a shell command? So just replacing `$prog` and `$args` and then passing the entire thing to `sh` (or however you get this binary in vscode).

Reply:
I think that it's not easy to make that solution portable across operating systems. The unix shells are pretty consistent with using `-c` for executing commands, but the same can't be said about Windows: cmd.exe uses `/C` and Powershell uses something like `Invoke-Command`. Figuring out the correct option for the user's selected shell seems like it will be too error-prone - especially considering that they could be using a shell that doesn't follow the platform conventions.

Using the shells as another layer will also make the error messages about the command fails harder to read.

Although it removes some abstraction, I think the best solution is to separate the `bin` and `args` in the configuration. The previous example would look like this:
```json
{
	"ocaml.sandbox": {
		"kind": "custom",
		"bin": "bash",
		"args": "-ic $prog $args"
	}
}
```

The `bin` field could have just the `$prog` substitution available, and the `args` field would have both the `$prog` and `$args` substitutions available.

Reply:
After some more investigation, passing the entire string from the args field as a single argument isn't satisfactory either. If I use my idea, `Window.createTerminal` breaks because it inserts quotes around args. Your idea does seem better.

I also found out that Node's child_process (what we use to execute all commands) has some [shell requirements](https://nodejs.org/api/child_process.html#child_process_shell_requirements) that we can take advantage of: 
> The shell should understand the -c switch. If the shell is 'cmd.exe', it should understand the /d /s /c switches and command line parsing should be compatible.

It turns out that powershell does understand `-c`, so the only special handling necessary is for cmd.exe

I will implement what you suggested.

Reply:
This has now been implemented with a slight change of terminology:
```json
"ocaml.sandbox": {
  "kind": "custom",
  "template": "bash -ic $prog $args"
}
```

Reply:
I just realized this doesn't handle spaces in the command name very well. Still needs more work...